From: Christoph Hellwig <hch@infradead.org>
Date: Sun, 8 Jul 2012 15:58:40 -0400
Subject: [PATCH] target: call transport_check_aborted_status from target_execute_cmd
Git-commit: d59a02b4ebe6fe5ac42b702f1ced6368ced78d76
References: FATE#313550
Patch-Mainline: v3.6

When we call target_execute_cmd for write commands the command has been
on the state list before an abort might have come in before
target_execute_cmd.  Call transport_check_aborted_status to deal with
this case.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Nicholas Bellinger <nab@linux-iscsi.org>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/target/target_core_transport.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/target/target_core_transport.c b/drivers/target/target_core_transport.c
index d9fe793..1c7021c 100644
--- a/drivers/target/target_core_transport.c
+++ b/drivers/target/target_core_transport.c
@@ -1865,6 +1865,12 @@ void target_execute_cmd(struct se_cmd *cmd)
 	struct se_device *dev = cmd->se_dev;
 
 	/*
+	 * If the received CDB has aleady been aborted stop processing it here.
+	 */
+	if (transport_check_aborted_status(cmd, 1))
+		return;
+
+	/*
 	 * Determine if IOCTL context caller in requesting the stopping of this
 	 * command for LUN shutdown purposes.
 	 */
-- 
1.7.4.2

